Runtimes/Unity/BepInEx.Unity.IL2CPP/Il2CppInteropManager.cs

```diff
@@ -282,25 +283,24 @@ private static void GenerateInteropAssemblies()
         }
     }
 
-    private static void DownloadUnityAssemblies()
-    {
+    private static void DownloadUnityAssemblies() {
         var unityVersion = UnityInfo.Version;
-        var source =
-            UnityBaseLibrariesSource.Value.Replace("{VERSION}",
-                                                   $"{unityVersion.Major}.{unityVersion.Minor}.{unityVersion.Build}");
-
-        if (!string.IsNullOrEmpty(source))
-        {
-            Logger.LogMessage("Downloading unity base libraries");
-
-            Directory.CreateDirectory(UnityBaseLibsDirectory);
-            Directory.EnumerateFiles(UnityBaseLibsDirectory, "*.dll").Do(File.Delete);
-
+        var file = $"{unityVersion.Major}.{unityVersion.Minor}.{unityVersion.Build}.zip";
+
+        var baseFolder = Directory.CreateDirectory(UnityBaseLibsDirectory);
+        baseFolder.EnumerateFiles("*.dll").Do(a=>a.Delete());
+        var target = baseFolder.GetFiles(file).FirstOrDefault();
+        if (target != null) {
+            using var fStream = target.OpenRead();
+            using var zipArchive = new ZipArchive(fStream, ZipArchiveMode.Read);
+            zipArchive.ExtractToDirectory(UnityBaseLibsDirectory);
```

Comment:
This could be deduplicated with the same code below.

Reply:
I'll end up with more lines that way because ZipArchive needs to be declared outside first. Then I need to add extra line to close the archive outside because I no longer can use "using". Using here makes this variable scoped so I can't extract it outside.

Reply:
Fair enough.